math/src/main/java/com/powsybl/math/matrix/SparseMatrix.java

```diff
@@ -242,8 +242,19 @@ public void iterateNonZeroValue(ElementHandler handler) {
     public void iterateNonZeroValueOfColumn(int j, ElementHandler handler) {
         int first = columnStart[j];
         if (first != -1) {
-            int last = j < columnStart.length - 1 ? columnStart[j + 1] : values.size();
-            for (int v = first; v < last; v++) {
+            // to find last column value, we have to go until next non empty column
```

Comment:
Note:
sometimes in sparse matrix representation, there is one more array for the element count in each column. It's a little less memory efficient but maybe simpler for iterations.

For readability, maybe we could have a dedicated method for finding the column count or the last index of a column ?

Reply:
Good remark, an additional array for column count is nothing on term of memory compare to values array in real data and also it would improve iteration on non zero value performance because in actual impl it requires to expore next columns to find last value.